ruby/generate_stubs.rb

```diff
@@ -0,0 +1,88 @@
+# This script parses the comments in the Ruby C extension files and translates them into Ruby
```

Comment:
Did you consider using https://docs.ruby-lang.org/en/3.4/RDoc/Parser/C.html to do the parsing? I ask because in addition to potentially lowering the maintenance burden through reuse, I think the comments in the existing C source files are constructed using the syntax it supports e.g. `call-seq`.

Reply:
My main concern with RDoc is that the results are pretty bare-bones - for example, there's no way to annotate the type of either inputs or outputs.